### PR TITLE
ci: use withMageEnv step

### PR DESCRIPTION
**Pull request by kuisathaverat**
Remove the Docker container we use to run Go and use native Go+mage by using `withMageEnv` step that manages to configure Go+Mmage.

### CODE REVIEW

**Review by v1v — APPROVED** [review]
💯

**Review by cachedout — APPROVED** [review]

**Review by axw — APPROVED** [review]

**Comment by axw** [comment]
@kuisathaverat are you waiting for us to merge this? Please feel free to merge when it's green.

**Pull request merged**
